socorro/external/postgresql/connection_context.py

```diff
@@ -68,9 +79,20 @@ def __init__(self, config, local_config=None):
             local_config - this is the namespace within the complete config
                            where the actual database parameters are found"""
         super(ConnectionContext, self).__init__()
+
         self.config = config
+
         if local_config is None:
             local_config = config
+
+        if local_config.get('database_url'):
+            url = urlparse.urlparse(local_config['database_url'])
+            local_config['database_username'] = url.username
+            local_config['database_password'] = url.password
+            local_config['database_hostname'] = url.hostname
+            local_config['database_port'] = url.port
+            local_config['database_name'] = url.path[1:]
+
```

Comment:
Hmm... what bothers me here is that we're taking these pieces and then create a string with it later (e.g `self.dsn = ...`)
So, if you leave the port out of the URL, which I think is a sensible thing to do. then `local_config['database_port'] is None` and when this is inserted into the `dsn` string you'll end up with `... port=None ...` which postgres won't like. 

Like we discussed in IRC, I don't think we should do `local_config['database_port'] = url.port or local_config['database_port']` because that's implicit and just odd. 
We could instead do: `local_config['database_port'] = url.port or 5432` but then we have to know that the default postgres port is 5432. 

How about...

We *can't* do `psycopg2.connect('postgres://user@password:localhost/mydb')` but you **can** do `psycopg2.connect(user="user", password="password", host="localhost", dbname="mydb")`
I.e. the equivalent for our use case would be `psycopg2.connect(**kwargs)`. 

A neat thing about this is that we can leave this as none and the psycopg2 will automatically know to set the default. This is also applicable not only to the port but to things like the username. 

[Full documentation here](http://initd.org/psycopg/docs/module.html)

cc @twobraids